### PR TITLE
PIN-3178 [BE] Spostare la validazione della risk analysis dalla bozza all'attivazione di una finalità

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala [hunk 1]
```diff
@@ -135,35 +136,63 @@ final case class PurposeApiServiceImpl(
     val operationLabel = s"Updating Purpose $purposeId"
     logger.info(operationLabel)
 
+    def isDraft(purpose: PurposeManagementDependency.Purpose): Boolean = {
+      val states = purpose.versions.map(_.state)
+      states == Seq(PurposeManagementDependency.PurposeVersionState.DRAFT)
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: This returns true only if exists just one version in draft state. Is this the requirement?

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: Well, just one version can be in draft, so the impl is not wrong imho

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala [hunk 2]
```diff
@@ -135,35 +136,63 @@ final case class PurposeApiServiceImpl(
...
     contexts: Seq[(String, String)],
     toEntityMarshallerPurpose: ToEntityMarshaller[Purpose],
     toEntityMarshallerProblem: ToEntityMarshaller[Problem]
   ): Route = authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE, M2M_ROLE) {
     val operationLabel = s"Retrieving Purpose $id"
     logger.info(operationLabel)
 
+    def tryToValidateRiskAnalysisForm(
+      riskAnalysisForm: Option[PurposeManagementDependency.RiskAnalysisForm]
+    ): Future[Option[Boolean]] = {
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: I would return a simple option here.
Wrapping in a Future is just for the for-compr aesthetic.

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala (hunk 1 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: > Well, just one version can be in draft, so the impl is not wrong imho

Nope.
`purpose.versions.map(_.state)` can contain more than one state and just one of them could be in draft.
So comparing with `Seq(PurposeManagementDependency.PurposeVersionState.DRAFT)`, return true only if exists one version in draft, but in the case of `Seq(PurposeManagementDependency.PurposeVersionState.ACTIVE, PurposeManagementDependency.PurposeVersionState.DRAFT)` it doesn't work.

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: What I'm trying to understand here is if using a `.exists` or `.contains` should be the right choice.

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: After some interactions with the team, I can confirm that this solution is fine.

File: src/test/scala/it/pagopa/interop/purposeprocess/PurposeVersionStateSpec.scala [hunk 3]
```diff
@@ -316,6 +316,41 @@ class PurposeVersionStateSpec extends AnyWordSpecLike with SpecHelper with Scala
       }
     }
 
+    "fail if risk analysis form is not valid" in {
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/purposeprocess/PurposeVersionStateSpec.scala]:
```suggestion
    "fail if risk analysis form is not valid on Version activation" in {
```

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala [hunk 4]
```diff
@@ -307,6 +307,12 @@ final case class PurposeApiServiceImpl(
       versionUUID    <- versionId.toFutureUUID
       organizationId <- getOrganizationIdFutureUUID(contexts)
       purpose        <- purposeManagementService.getPurpose(purposeUUID)
+      riskAnalysisForm = purpose.riskAnalysisForm.map(RiskAnalysisConverter.dependencyToApi)
+      _              <- riskAnalysisForm
+        .traverse(RiskAnalysisValidation.validate(_))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: converting back and forth works but it's not optimal

Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala]: It is better to open a fresh new story to accomplish this goal.

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**